frontend/vite-project/src/stores/shippingStore.ts

```diff
@@ -0,0 +1,28 @@
+import { create } from 'zustand';
+import { ShippingFormTypes } from '../types/shippingFormTypes'
+
+interface ShippingState {
+  formData: ShippingFormTypes;
+  setFormData: (data: ShippingFormTypes) => void;
+  updateFormField: <K extends keyof ShippingFormTypes>(key: K, value: ShippingFormTypes[K]) => void;
+}
+
+export const useShippingStore = create<ShippingState>((set) => ({
+  formData: {
+    firstName: '',
+    lastName: '',
+    address: '',
+    apartment: '',
+    postalCode: '',
+    city: '',
+    saveContact: false,
+  },
+  setFormData: (data) => set({ formData: data }),
+  updateFormField: (key, value) =>
+    set((state) => ({
+      formData: {
+        ...state.formData,
+        [key]: value,
+      },
+    })),
+}));
```

Comment:
```suggestion
}));
export const useFormData = () => useShippingStore((state) => state.formData);
export const useShippingActions = () => useShippingStore((state) => state.actions);
```
Export custom hooks for cleaner and more modular access to the shipping store
 useFormData returns the current form data state.
 useShippingActions provides access to all state-modifying actions.